filters/tracing/statebagtotag.go

```diff
@@ -57,6 +58,15 @@ func (f stateBagToTagFilter) Request(ctx filters.FilterContext) {
 	if span == nil {
 		return
 	}
+
+	if f.stateBagItemName == logfilter.AuthUserKey {
```

Comment:
Is this special treatment really necessary? 
Won't
```
stateBagToTag("masked-auth-user", "client_id")
```
just work?

Reply:
yes, but then we have to make sure that nobody uses `stateBagToTag("auth-user", "client_id")`

Reply:
Oh, I see now -  it is not a feature to allow user to set tags only for certain auth realms but rather mask uids even though user wants `stateBagToTag("auth-user", "client_id")`, i.e. they will get realm instead of uid in masked realms.

Still it should be cleaner to persist realm unconditionally into statebag in tokeninfo filter and concentrate all logic of realm match and uid->realm substitution in the `stateBagToTag` filter.

It also changes simple `put this there` semantics of `stateBagToTag` into `put this there unless it is not auth-user and realm matches commandline switch regexp - in such a case substitute auth-user to ream` o_O